entrypoint.sh

```
@@ -2,7 +2,7 @@
 
 cp /action/problem-matcher.json /github/workflow/problem-matcher.json
 
-git clone --depth 1 -b 2.3.0 https://github.com/WordPress/WordPress-Coding-Standards.git ~/wpcs
+composer global require "dealerdirect/phpcodesniffer-composer-installer:^0.7.1" "wp-coding-standards/wpcs:^3.1.0"
```

@sksaju As noted in @iamdharmesh's run log https://github.com/iamdharmesh/autoshare-for-twitter/actions/runs/10319895557/job/28569247554?pr=5 this workflow doesn't work if `composer` isn't available.

Since this PR adds a requirement for `composer` to exist in the testrunner's `$PATH`, this PR should also:

- `set -ueo pipefail` to cause the check to fail early if `composer` is not installed
- add a note in `CHANGELOG.md` that this tool now requires `composer` in the testrunner

Alternatively:

- replace the `composer global require` with a different method of installing PHPCS and the WordPress Coding Standards

---

@iamdharmesh any preference on approach noted by @benlk above?

---

Thanks for checking on this and sharing approaches, @benlk. @jeffpaul, the approaches shared by Ben looks good, and I would prefer handling the `composer` installation in the action itself (in the `Dockerfile`), so action users don’t have to deal with it.

---

So then in https://github.com/10up/wpcs-action/blob/05677bafa9c8c199acecfe79fd464d69fd6dbf0e/Dockerfile#L11 we might add `apk add composer`

---

Thanks, @iamdharmesh @benlk, I've added the composer installation script in the PR, could you take another look, please?

---

Thanks for making the changes, @sksaju. The code looks good. However, I tried running the action on the repo, and it’s still failing due to a missing required PHP extension. I didn't got the time to look into it deeply, but here are the logs: https://github.com/iamdharmesh/autoshare-for-twitter/actions/runs/10852960407/job/30120164745?pr=5

Thank you.

---

In that build log, there are two notes about missing php extensions:

>     - wp-coding-standards/wpcs 3.1.0 requires ext-tokenizer * -> it is missing from your system. Install or enable PHP's tokenizer extension.

>     - squizlabs/php_codesniffer[2.0.0, ..., 2.9.2] require ext-tokenizer * -> it is missing from your system. Install or enable PHP's tokenizer extension.
>    - squizlabs/php_codesniffer[3.8.0, ..., 3.10.2] require ext-simplexml * -> it is missing from your system. Install or enable PHP's simplexml extension.
    
I expect that `ext-tokenizer` and `ext-simplexml` are available in the `apk` package repository.

Looking at the results of `apk update &&     apk upgrade &&     apk add git &&     apk add composer` earlier in that build log, it looks like we're using `php8-` on Alpine 3.16, and [in the list of packages matching that constraint](https://pkgs.alpinelinux.org/packages?name=php8-*&branch=v3.16&repo=community&arch=&maintainer=), we would use the packages `php8-simplexml` and `php8-tokenizer` 

To do:
- [ ] `apk add php8-simplexml`
- [ ] `apk add php8-tokenizer`
- [ ] since `apk add` supports multiple packages in one call according to https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper#Add_a_Package , condense the `apk add` calls to `apk add git composer php8-simplexml php8-tokenizer` and anything else that needs to be installed.
- [ ] if mere installation doesn't fix the issue, see if we need to adjust `php.ini` to enable those extensions.

---

@benlk I added the required extensions but looks like there is some other issue.

This is the latest error `ERROR: Referenced sniff "PHPCSUtils" does not exist`

but phpcsutils seems to be installed? 
https://github.com/thrijith/autoshare-for-twitter/actions/runs/10938472447/job/30369473907?pr=2#step:4:46
https://github.com/thrijith/autoshare-for-twitter/actions/runs/10938472447/job/30369473907?pr=2#step:4:40

---

Installed, but not in `installed_paths` for some reason. If we explicitly set `installed_paths`, does the `composer`-installed standard's automatic registration with PHPCS get overridden? I'll make another comment with the needed changes.